tf2_eigen/include/tf2_eigen/tf2_eigen.h

```diff
@@ -186,10 +314,42 @@ geometry_msgs::msg::Pose toMsg(const Eigen::Affine3d& in) {
   msg.position.x = in.translation().x();
   msg.position.y = in.translation().y();
   msg.position.z = in.translation().z();
-  msg.orientation.x = Eigen::Quaterniond(in.rotation()).x();
-  msg.orientation.y = Eigen::Quaterniond(in.rotation()).y();
-  msg.orientation.z = Eigen::Quaterniond(in.rotation()).z();
-  msg.orientation.w = Eigen::Quaterniond(in.rotation()).w();
+  Eigen::Quaterniond q(in.linear());
```

Comment:
If that's possible, flip the quaternion in a single line operation, instead of manually doing it below.
```suggestion
  Eigen::Quaterniond q(in.linear());
  if (q.w() < 0)
    q *= 1.0;
```

Reply:
I'm not sure that we want to be doing another operation if we can avoid it. Every extra operation is potentially a loss of precision as well as costing resources. 

The negative w values are completely valid quaternions. I'd rather not just normalize it here. This breaks the expectation that fromMsg(toMsg(...)) will return the same value.